sim/core/dot.go

```diff
@@ -383,3 +383,30 @@ func (spell *Spell) createDots(config DotConfig, isHot bool) {
 		}
 	}
 }
+
+func (dot *Dot) CopyDotAndApply(sim *Simulation, originaldot *Dot) {
+	dot.TakeSnapshot(sim, false)
+	dot.SnapshotBaseDamage = originaldot.SnapshotBaseDamage
+
+	dot.tickPeriod = originaldot.tickPeriod
+	dot.NumberOfTicks = originaldot.NumTicksRemaining(sim) // originaldot.NumberOfTicks
```

Comment:
Changing `NumberOfTicks` is a destructive action. You should not change this but instead change the `TickCount` to match the count of the original dot

Reply:
Did some testing and it didn't seem necessary to even have this line. Removed for now.